packages/web/src/components/map-layout.tsx

```diff
@@ -0,0 +1,85 @@
+import GoogleMapReact from "google-map-react";
+import React from "react";
+const markers = [];
+let labelIndex = 1;
+const deleteMarker = (marker, markers) => {
+    markers.splice(markers.indexOf(marker), 1);
+    marker.setMap(null);
+    marker = null;
+    markers.forEach((temp) => {
+        temp.setLabel(String(markers.indexOf(temp) + 1));
+    });
+    labelIndex = markers.length + 1;
+};
+const addMarker = (location, map, maps) => {
+    const marker = new maps.Marker({
+        position: location,
+        map: map,
+        title: "test",
+        label: String(labelIndex),
+    });
+    maps.event.addListener(marker, "rightclick", function() {
+        deleteMarker(this, markers);
+    });
+    markers.push(marker);
+};
+
+/*const resetBounds = (map, maps) => {
+    //Create new bounds object
+    const bounds = new maps.LatLngBounds();
+    //Loop through an array of points, add them to bounds
+    for (let i = 0; i < data.length; i++) {
+        const geoCode = new maps.LatLng(data[i][1], data[i][2]);
+        bounds.extend(geoCode);
+    }
+    //Add new bounds object to map
+    map.fitBounds(bounds);
+};*/
+const handleApiLoaded = (map, maps) => {
```

Comment:
Are there types for the arguments from `google-map-react`? This function especially since it's used in the `onGoogleApiLoaded` prop

Reply:
Looking at the node_modules/@types/google-map-react/index.d.ts file, I couldn't recognize which type to use for the functions outside the class, so I used direct references to the Google Maps Internal API. Using onGoogleApiLoaded allows me to use the internal API, as I didn't know how to use the existing types to create functions that will let me edit marker's props and manage multiple markers. I don't know how to give the vars their proper types since I used the internal API.

Reply:
Got it, yeah I see there's no internal maps types for `google-map-react`